File: be/src/pos-terminals/dto/pos-terminal.dto.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
import { PaymentMethodEntryDto } from './payment-method-entry.dto';
import type { PosTerminal } from '../entities/pos-terminal.entity';

export class PosTerminalDto {
  @ApiProperty({ example: 1 })
  id: number;

  @ApiProperty({ example: '018f2c3d-...' })
  kioskId: string;

  @ApiProperty({ example: '123 Main St.' })
  address: string;

  @ApiPropertyOptional({ example: 'ABC Corporation' })
  legalName: string | null;

  @ApiProperty({ example: '123-456-789-000' })
  tinNumber: string;

  @ApiProperty({ type: [PaymentMethodEntryDto] })
  paymentMethods: PaymentMethodEntryDto[];

  static from(terminal: PosTerminal): PosTerminalDto {
    const dto = new PosTerminalDto();
    dto.id = terminal.id;
    dto.kioskId = terminal.kioskId;
    dto.address = terminal.address;
    dto.legalName = terminal.legalName;
    dto.tinNumber = terminal.tinNumber;
    dto.paymentMethods = (terminal.paymentMethods ?? []).map(PaymentMethodEntryDto.from);
    return dto;
  }
}

File: kiosk/lib/data/backend_api/schemas/pos_terminal_dto.dart
import 'package:dart_mappable/dart_mappable.dart';

import 'payment_method_entry_dto.dart';

part 'pos_terminal_dto.mapper.dart';

@MappableClass()
class PosTerminalDto with PosTerminalDtoMappable {
  const PosTerminalDto({
    required this.id,
    required this.kioskId,
    this.legalName,
    required this.address,
    required this.tinNumber,
    required this.paymentMethods,
  });

  final int id;
  final String kioskId;
  final String? legalName;
  final String address;
  final String tinNumber;
  final List<PaymentMethodEntryDto> paymentMethods;

  static const fromJson = PosTerminalDtoMapper.fromJson;
}
